components/builder-jobsrv/src/test_helpers.rs

```diff
@@ -0,0 +1,192 @@
+}
+
+impl DbHelper {
+    pub fn new(group_id: i64, target: &str) -> Self {
+        DbHelper { group_id,
+                   target: BuilderPackageTarget(PackageTarget::from_str(target).unwrap()),
+                   name_map: HashMap::new(),
+                   id_map: HashMap::new() }
+    }
+
+    pub fn add(&mut self,
+               conn: &diesel::pg::PgConnection,
+               name: &str,
+               deps: &[&str],
+               job_state: JobExecState)
+               -> i64 {
+        let dependencies: Vec<i64> =
+            deps.iter()
+                .map(|d| {
+                    *(self.name_map
+                          .get(d.to_owned())
+                          .unwrap_or_else(|| panic!("Dependency {} not found", d)))
+                })
+                .collect();
+
+        // let plan_name = name.split('/').take(2).collect::<Vec<&str>>().join("/");
+        // TODO lookup/create project_id based on plan_name
+
+        let entry = NewJobGraphEntry { group_id: self.group_id,
+                                       job_state,
+                                       project_name: "dummy_project_id",
+                                       job_id: None,
+                                       manifest_ident: name,
+                                       as_built_ident: None,
+                                       dependencies: &dependencies,
+                                       waiting_on_count: dependencies.len() as i32,
+                                       target_platform: self.target };
+
+        let job_graph_entry = JobGraphEntry::create(&entry, &conn).unwrap();
+
+        self.name_map.insert(name.to_owned(), job_graph_entry.id);
+        self.id_map.insert(job_graph_entry.id, name.to_owned());
+        job_graph_entry.id
+    }
+
+    pub fn id_by_name(&self, name: &str) -> i64 {
+        *(self.name_map
+              .get(name)
+              .unwrap_or_else(|| panic!("No entry for {}", name)))
+    }
+
+    #[allow(dead_code)]
+    pub fn name_by_id(&self, id: i64) -> String {
+        self.id_map
+            .get(&id)
+            .unwrap_or_else(|| panic!("No entry for {}", id))
+            .clone()
+    }
+}
+
+pub fn make_simple_graph_helper(group_id: i64,
+                                target_platform: &BuilderPackageTarget,
+                                conn: &diesel::pg::PgConnection)
+                                -> DbHelper {
+    let mut h = DbHelper::new(group_id, target_platform);
+
+    h.add(conn, "foo/bar/1.2.3/123", &[], JobExecState::Ready);
+    h.add(conn,
+          "foo/baz/1.2.3/123",
+          &["foo/bar/1.2.3/123"],
+          JobExecState::WaitingOnDependency);
+    h.add(conn,
+          "foo/ping/1.2.3/123",
+          &["foo/bar/1.2.3/123"],
+          JobExecState::WaitingOnDependency);
+    h.add(conn,
+          "foo/pong/1.2.3/123",
+          &["foo/baz/1.2.3/123", "foo/ping/1.2.3/123"],
+          JobExecState::WaitingOnDependency);
+
+    h
```

Comment:
For these tests, I wonder if it would be worth creating some kind of file format (probably inspired by `dot` syntax) so that you could declare several graphs as readable test fixture files. (If it were based on `dot` syntax, you might even be able to generate visual representations of them, which would be fantastic for understanding!) Then your test helpers could just have general logic for reading these files and generating the code that represents them for tests.